--- yp-tools-2.7/src/Makefile.am~	Wed Jun  5 14:40:19 2002
+++ yp-tools-2.7/src/Makefile.am	Wed Jun  5 14:40:19 2002
@@ -6,27 +6,21 @@
 AUTOMAKE_OPTIONS = 1.4 gnits
 MAINT_CHARSET = latin1
 #
-INSTALL_PROGRAM = @INSTALL_PROGRAM@ -s -m 555
 
-CFLAGS = @WARNFLAGS@ @DEFS@ @CFLAGS@
-LDADD = @LIBINTL@ ../lib/libyptools.a
+LDADD = $(INTLLIBS) ../lib/libyptools.a
 
 localedir = $(datadir)/locale
 INCLUDES = -I$(srcdir) -I$(top_srcdir) -DLOCALEDIR=\"$(localedir)\"
 
 CLEANFILES = *~ domainname
 
-rootexecbindir = @rootbindir@
-
 EXTRA_PROGRAMS = domainname
-rootexecbin_PROGRAMS = @DOMAINNAME@
+rootbin_PROGRAMS = $(DOMAINNAME)
 bin_PROGRAMS = ypwhich ypmatch ypcat yppasswd
 sbin_PROGRAMS = yppoll ypset yptest
 
-domainname_LDADD := @LIBINTL@
-
 yppasswd_SOURCES = yppasswd.c yppasswd_xdr.c yppasswd.h
-yppasswd_LDADD = ${LDADD} @LIBCRYPT@ @LIBCRACK@
+yppasswd_LDADD = ${LDADD} $(LIBCRYPT) $(LIBCRACK)
 yppasswd_CFLAGS = -DPASSWD_PROG=\"${PASSWD_PROG}\" \
 	-DCHFN_PROG=\"${CHFN_PROG}\" -DCHSH_PROG=\"${CHSH_PROG}\"
 
--- yp-tools-2.7/man/Makefile.am~	Wed Jun  5 14:42:15 2002
+++ yp-tools-2.7/man/Makefile.am	Wed Jun  5 14:42:15 2002
@@ -14,4 +14,4 @@
 	yppoll.8.in ypset.8.in yptest.8.in ypwhich.1.in nicknames.5.in
 
 man_MANS = ypcat.1 ypchfn.1 ypchsh.1 ypmatch.1 yppasswd.1 yppoll.8 ypset.8\
-	 yptest.8 ypwhich.1 nicknames.5 @MAN_DOMAINNAME@
+	 yptest.8 ypwhich.1 nicknames.5 $(MAN_DOMAINNAME)
--- yp-tools-2.7/lib/Makefile.am~	Wed Jun  5 14:44:06 2002
+++ yp-tools-2.7/lib/Makefile.am	Wed Jun  5 14:44:06 2002
@@ -10,14 +10,11 @@
 
 noinst_LIBRARIES = libyptools.a
 
-CFLAGS = @WARNFLAGS@ @DEFS@ @CFLAGS@
-
 EXTRA_DIST = getopt.c getopt1.c getline.c ypbinderr_string.c
 
 noinst_HEADERS = getopt.h getline.h yp-tools.h nicknames.h yp_all_host.h
 
 libyptools_a_SOURCES = nicknames.c yp_xdr.c yp_all_host.c
-libyptools_a_LIBADD = @LIBOBJS@
 
 localedir = $(datadir)/locale
 INCLUDES = -I../intl -DLOCALEDIR=\"$(localedir)\"
--- yp-tools-2.7/Makefile.am~	Wed Jun  5 14:45:06 2002
+++ yp-tools-2.7/Makefile.am	Wed Jun  5 14:45:06 2002
@@ -6,12 +6,10 @@
 AUTOMAKE_OPTIONS = 1.5 gnits dist-bzip2
 MAINT_CHARSET = latin1
 #
-SUBDIRS = m4  lib src po man etc
+SUBDIRS = intl lib src po man etc
 
 CLEANFILES = *~
 
 #EXTRA_DIST = yp-tools.spec
 
-ACLOCAL_AMFLAGS = -I m4
-
 EXTRA_DIST = config.rpath
--- yp-tools-2.7/acinclude.m4~	Wed Jun  5 14:45:54 2002
+++ yp-tools-2.7/acinclude.m4	Wed Jun  5 14:45:54 2002
@@ -2,7 +2,7 @@
 dnl We can't just do AC_REPLACE_FUNCS(getline) because some systems
 dnl have a function by that name in -linet that doesn't have anything
 dnl to do with the function we need.
-AC_DEFUN(AM_FUNC_GETLINE,
+AC_DEFUN([AM_FUNC_GETLINE],
 [dnl
   am_getline_needs_run_time_check=no
   AC_CHECK_FUNC(getline,
@@ -46,7 +46,7 @@
 
 # serial 1
 
-AC_DEFUN(AM_WITH_EFENCE,
+AC_DEFUN([AM_WITH_EFENCE],
 [AC_MSG_CHECKING(if malloc debugging with efence is wanted)
 AC_ARG_WITH(efence,
 [  --with-efence           use ElectricFence 2.05],
--- yp-tools-2.7/configure.in~	Wed Jun  5 17:43:35 2002
+++ yp-tools-2.7/configure.in	Wed Jun  5 17:43:35 2002
@@ -4,78 +4,62 @@
 AM_CONFIG_HEADER(config.h)
 AC_PREFIX_DEFAULT(/usr)
 
-dnl Set of available languages.
-ALL_LINGUAS="de"
-
-AC_SUBST(PACKAGE)
-AC_SUBST(VERSION)
-
 dnl Some hacks...
-test "${prefix}" = "NONE" && prefix="/usr"
-if test ${prefix} = '/usr'
-then
-dnl If we use /usr as prefix, use /usr/share/man for manual pages
-	if test ${mandir} = '${prefix}/man'
-        then
-           mandir='${prefix}/share/man'
-        fi
+if test "${prefix}" = "NONE"; then
+	prefix="/usr"
 fi
 dnl Change libexec to lib/yp
-if test ${libexecdir} = '${exec_prefix}/libexec'
-then
-  libexecdir='${exec_prefix}/lib/yp'
+if test ${libexecdir} = '${exec_prefix}/libexec'; then
+	libexecdir='${exec_prefix}/lib/yp'
 fi
 
-if test x"$ac_default_prefix" = x"/usr"
-then
-   rootbindir="/bin"
+if test x"$ac_default_prefix" = x"/usr"; then
+	rootbindir="/bin"
 else
-   rootbindir=$ac_default_prefix"/bin"
+	rootbindir=$ac_default_prefix"/bin"
 fi
 AC_SUBST(rootbindir)
 
 dnl Should we install domainname ? default yes
 AC_ARG_ENABLE(domainname,
-        [  --disable-domainname    Don't install domainname [default=yes]],
-              INSTALL_DOMAINNAME=$enableval, INSTALL_DOMAINNAME=yes)
-if test "$INSTALL_DOMAINNAME" = "no"
-then
-   DOMAINNAME=
-   MAN_DOMAINNAME=
+	[  --disable-domainname    Don't install domainname [default=yes]],
+		INSTALL_DOMAINNAME=$enableval, INSTALL_DOMAINNAME=yes)
+if test "$INSTALL_DOMAINNAME" = "no"; then
+	DOMAINNAME=
+	MAN_DOMAINNAME=
 else
-   DOMAINNAME=domainname
-   MAN_DOMAINNAME="domainname.8 nisdomainname.8 ypdomainname.8"
+	DOMAINNAME=domainname
+	MAN_DOMAINNAME="domainname.8 nisdomainname.8 ypdomainname.8"
 fi
 AC_SUBST(DOMAINNAME)
 AC_SUBST(MAN_DOMAINNAME)
 
 dnl Should we use the CRACKLIB ? default no
 AC_ARG_ENABLE(cracklib,
-        [  --enable-cracklib       Use the cracklib for yppasswd [default=no]],
-        USE_CRACKLIB=1
-        AC_DEFINE(USE_CRACKLIB,1,[Define if you want to use libcrack to check the new passwords]))
+	[  --enable-cracklib       Use the cracklib for yppasswd [default=no]],
+	USE_CRACKLIB=1
+	AC_DEFINE(USE_CRACKLIB,1,[Define if you want to use libcrack to check the new passwords]))
 
 dnl Should we do extra checks even if we use the CRACKLIB ? default no
 AC_ARG_ENABLE(cracklib-strict,
-        [  --enable-cracklib-strict Do some extra tests before accepting [default=no]],
-        USE_CRACKLIB=1;
-        USE_CRACKLIB_STRICT=1;
-        AC_DEFINE(USE_CRACKLIB,1,[Define if you want to use libcrack to check the new passwords])
-        AC_DEFINE(USE_CRACKLIB_STRICT,1,[Define if you want to use some extra checks besides the cracklib]))
+	[  --enable-cracklib-strict Do some extra tests before accepting [default=no]],
+	USE_CRACKLIB=1;
+	USE_CRACKLIB_STRICT=1;
+	AC_DEFINE(USE_CRACKLIB,1,[Define if you want to use libcrack to check the new passwords])
+	AC_DEFINE(USE_CRACKLIB_STRICT,1,[Define if you want to use some extra checks besides the cracklib]))
 
 dnl Should passwd be used instead of yppasswd ? Default is no.
 AC_ARG_ENABLE(call-passwd,
-        [  --enable-call-passwd  yppasswd calls passwd, chfn or chsh [default=no]],
+	[  --enable-call-passwd  yppasswd calls passwd, chfn or chsh [default=no]],
 	YPPASSWD_IS_DEPRECATED=1
-        AC_DEFINE(YPPASSWD_IS_DEPRECATED,1,[Define if yppasswd is deprecated and passwd should be used instead]))
-
+	AC_DEFINE(YPPASSWD_IS_DEPRECATED,1,[Define if yppasswd is deprecated and passwd should be used instead]))
 
 dnl Checks for programs.
 AC_PROG_CC
 dnl If we're using gcc, we want warning flags
-dnl test -n "$GCC" && WARNFLAGS="-Wall"
-test -n "$GCC" && WARNFLAGS="-W -Wall -Wbad-function-cast -Wcast-align -Winline -Wmissing-declarations -Wmissing-prototypes -Wnested-externs -Wshadow -Wstrict-prototypes -Wundef"
-AC_SUBST(WARNFLAGS)
+if test -n "$GCC"; then
+	CFLAGS="$CFLAGS -W -Wall -Wbad-function-cast -Wcast-align -Winline -Wmissing-declarations -Wmissing-prototypes -Wnested-externs -Wshadow -Wstrict-prototypes -Wundef"
+fi
 AC_PROG_CPP
 AC_ISC_POSIX
 AC_PROG_INSTALL
@@ -83,9 +67,9 @@
 AC_PROG_RANLIB
 
 if test "${YPPASSWD_IS_DEPRECATED}" = "1"; then
-  AC_PATH_PROG(PASSWD_PROG, passwd)
-  AC_PATH_PROG(CHFN_PROG, chfn)
-  AC_PATH_PROG(CHSH_PROG, chsh)
+	AC_PATH_PROG(PASSWD_PROG, passwd)
+	AC_PATH_PROG(CHFN_PROG, chfn)
+	AC_PATH_PROG(CHSH_PROG, chsh)
 fi
 
 AM_WITH_DMALLOC
@@ -96,18 +80,18 @@
 AC_CHECK_LIB(nsl, yp_get_default_domain)
 
 AC_CHECK_LIB(xcrypt,crypt,LIBCRYPT="-lxcrypt",LIBCRYPT="")
-if test -z "$LIBCRYPT" ; then
-  AC_CHECK_LIB(crypt,crypt,LIBCRYPT="-lcrypt",LIBCRYPT="")
-  AC_CHECK_HEADERS(crypt.h)
+if test -z "$LIBCRYPT"; then
+	AC_CHECK_LIB(crypt,crypt,LIBCRYPT="-lcrypt",LIBCRYPT="")
+	AC_CHECK_HEADERS(crypt.h)
 else
-  AC_CHECK_HEADERS(xcrypt.h crypt.h)
+	AC_CHECK_HEADERS(xcrypt.h crypt.h)
 fi
 AC_SUBST(LIBCRYPT)
 
 dnl Check for libcrack
 if test "${USE_CRACKLIB}" = "1"; then
-  AC_CHECK_LIB(crack, FascistCheck, LIBCRACK="-lcrack",LIBCRACK="")
-  AC_SUBST(LIBCRACK)
+	AC_CHECK_LIB(crack, FascistCheck, LIBCRACK="-lcrack",LIBCRACK="")
+	AC_SUBST(LIBCRACK)
 fi
 
 dnl Checks for header files.
@@ -118,21 +102,36 @@
 AC_C_CONST
 
 dnl Checks for library functions.
-AC_CHECK_FUNC(getopt_long, , [AC_LIBOBJ(getopt) AC_LIBOBJ(getopt1)])
+AC_CHECK_FUNC(getopt_long, [], [AC_LIBOBJ(getopt) AC_LIBOBJ(getopt1)])
 
 AC_REPLACE_FUNCS(ypbinderr_string)
 
 AM_FUNC_GETLINE
 if test $am_cv_func_working_getline != yes; then
-  AC_CHECK_FUNCS(getdelim)
+	AC_CHECK_FUNCS(getdelim)
 else
-  AC_CHECK_FUNCS(getline)
+	AC_CHECK_FUNCS(getline)
 fi
 
 dnl internationalization macros
-AM_GNU_GETTEXT([external])
+ALL_LINGUAS="de"
+AM_GNU_GETTEXT
 
-AC_OUTPUT([Makefile lib/Makefile src/Makefile man/Makefile m4/Makefile
-	   etc/Makefile po/Makefile.in man/domainname.8 man/nicknames.5
-	   man/ypcat.1 man/ypmatch.1 man/yppasswd.1 man/yppoll.8
-	   man/ypset.8 man/yptest.8 man/ypwhich.1])
+AC_OUTPUT([
+	Makefile 
+	intl/Makefile
+	lib/Makefile
+	src/Makefile
+	man/Makefile
+	etc/Makefile
+	po/Makefile.in
+	man/domainname.8
+	man/nicknames.5
+	man/ypcat.1
+	man/ypmatch.1
+	man/yppasswd.1
+	man/yppoll.8
+	man/ypset.8
+	man/yptest.8
+	man/ypwhich.1
+	])
